src/client/components/ProductDetails/ProductDetails.component.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import NumberInput from '../NumberInput/NumberInput.component'
+import ButtonComponent from '../Button/Button.component'
+import './ProductDetails.style.css'
+import defaultImage from '../../assets/images/image01.png'
+
+
+export const ProductDetails = (
+    {
+        imgSource,
+        ProductName,
+        RemainingUnit,
+        Price,
+        productColor,
+        productSize,
+        onClick,
+    }
+) => {
+    const [iconSource, setIconSource] = React.useState(true)
```

Comment:
Consider adding props for getting the state of the heart and for setting it outside of the component, so it is prepared for getting the state from the backend.